hands-on-exercise/2-ignite-cli-adv/6-gas-meter.md

```diff
@@ -89,30 +89,27 @@ Add a line that consumes or refunds the designated amount of gas in each relevan
 
 2. When handling a move:
 
-    ```diff-go [https://github.com/cosmos/b9-checkers-academy-draft/blob/gas-meter/x/checkers/keeper/msg_server_play_move.go#L88]
+    ```diff-go [https://github.com/cosmos/b9-checkers-academy-draft/blob/gas-meter/x/checkers/keeper/msg_server_play_move.go#L90]
         ...
         k.Keeper.SetSystemInfo(ctx, systemInfo)
     +  ctx.GasMeter().ConsumeGas(types.PlayMoveGas, "Play a move")
         ...
     ```
 
-3. When handling a game rejection, you make sure that you are not refunding more than what has already been consumed:
+Note that when handling a move, the gas is measured towards the end. This is a design decision, and you may arrive at a different conclusion. Indeed, if a player hits an error, such as a failure to pay the wager, or passing an illegal move, then the code will return an error before the gas metering line. This means this _errored_ player will pay only the gas already metered by the rest of the Cosmos SDK. If you decide to charge extra gas for a move whatever the errors, then you would move the gas metering line closer to the beginning of the function's body.
```

Comment:
@xavierlepretre 
> This means this _errored_ player will pay only the gas already metered by the rest of the Cosmos SDK.

This refers to _the player **making** the error_,  correct?

Reply:
> This refers to _the player **making** the error_, correct?

Yes

Reply:
> This means, in effect, "charging gas for moves that are not accepted due to errors", yes?

Both our versions are too confusing. The true / false pivot here is:

> the gas is measured towards the end

* If this is true (as currently implemented), then -> This means this _errored_ player will pay only the gas already metered by the rest of the Cosmos SDK
* If this is false (as is up to the developer), then the gas would be measure towards the beginning of the function, then that's equivalent to -> you decide to charge extra gas for a move whatever the errors

Reply:
Thanks for this :-) my review push will include a revision on this point, so I'll add a note then to draw attention.